spec/controllers/goals_controller_spec.rb

```diff
@@ -30,4 +30,70 @@
       expect(response).to redirect_to(user_path(my_user.id))
     end
   end
+
+  describe "DELETE #destroy" do
+    it "responds with http status 200 ok" do
+      user = create(:user)
+      goal = create(:goal, user: user)
+
+      sign_in(user)
+
+      delete :destroy, format: :js, id: goal.id
+      expect(response).to have_http_status(200)
+    end
+
+    it "deletes the expected goal" do
+      user = create(:user)
+      goal = create(:goal, user: user)
+
+      sign_in(user)
+
+      delete :destroy, format: :js, id: goal.id
+      expect(Goal.count).to eq(0)
+    end
+
+    it "deletes only the expected goal" do
+      user = create(:user)
+      goal = create(:goal, user: user)
+      second_goal = create(:goal, user: user)
+
+      sign_in(user)
+
+      delete :destroy, format: :js, id: goal.id
+      expect(Goal.count).to eq(1)
+    end
+
+    it "does not delete goal belonging to another user" do
+      user = create(:user)
+      goal = create(:goal, user: user)
+      other_user = create(:user, email: "someone.else@gmail.com")
+
+      sign_in(other_user)
+
+      delete :destroy, format: :js, id: goal.id
+      expect(Goal.count).to eq(1)
+    end
+
+    xit "sets the successful flash message" do
+      user = create(:user)
+      goal = create(:goal, user: user)
+
+      sign_in(user)
+
+      delete :destroy, format: :js, id: goal.id
+      expect(page).to have_content("Goal completed successfully!")
+    end
+
+    xit "responds with http status 404 not found for unkown goal" do
```

Comment:
> Is there a way to have RSpec expect a line to fail?

Yes, code towards end of this comment. Here's some context first.

When a controller action raises the `ActiveRecord::RecordNotFound` error, Rails (when running as `rails s`) will automatically catch that error and send the 404 Not Found response to the user. This behaviour is usually what we want so its good that Rails does this for us.

What I forgot when we discussed this previously is this `ActiveRecord::RecordNotFound`-error-to-404 translation isn't testable from within a controller spec. 

Rails performs the translation in code outside of the controllers that is not testable within a controller spec.

As we can't test the translation in the controller spec, it's good enough to test that the `ActiveRecord::RecordNotFound` error is raised. If a controller action raises that error, then its safe to assume that it will result in a 404 Not Found response being sent to the user.

To test the error is raised, use RSpec's `raise_error` matcher ([API docs and examples](https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/raise-error-matcher)):

```ruby
expect do
  delete :destroy, format: :js, id: goal.id
end.to raise_error(ActiveRecord::RecordNotFound)
```